### PR TITLE
Initial Filechooser modal

### PR DESCRIPTION
**Pull request by nidhibhatg**
## What changes were proposed in this pull request?

Added a modal in the importer page as a initial file chooser component

### CODE REVIEW

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from 'react';
+import './FileChooserComponentButton.scss';
+import FileChooserModal from '../FileChooserModal/FileChooserModal';
+import { Button } from 'antd';
+
+const FileChooserComponentButton = (): JSX.Element => {
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx:6]: This component actually includes the whole FileChooserModal so calling it a button could be a bit misleading, for now perhaps we can call it `FileChooserWithButton` (there is no need to include "component" in any of the component or folder names).

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from 'react';
+import './FileChooserComponentButton.scss';
+import FileChooserModal from '../FileChooserModal/FileChooserModal';
+import { Button } from 'antd';
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx:4]: nit: We can group the import statements with external libraries first and then our internal imports

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const FileChooserComponentButton = (): JSX.Element => {
+  const [show, setShow] = useState(false);
+
+  return (
+    <div>
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx:10]: Unless you actually need this div you can remove it and use `<>...</>` instead

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      <Button type="primary" onClick={() => setShow(true)}>
+        File chooser component
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx:12]: I can imagine that we wanna reuse this button in multiple places, and possibly use different labels for the button, so you can add a prop this string. (In my upcoming i18n PR I will show how to combine i18n with props and default values)

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from 'react';
+import './FileChooserComponentButton.scss';
+import FileChooserModal from '../FileChooserModal/FileChooserModal';
+import { Button } from 'antd';
+
+const FileChooserComponentButton = (): JSX.Element => {
+  const [show, setShow] = useState(false);
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserComponentButton/FileChooserComponentButton.tsx:7]: As we move forward with the integration we probably want to lift the state into the page/component that actually uses the new FileChoser and we will be needing callbacks to handle the actions triggered by this component, but for now we can leave it.

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  show: boolean;
+  onClose: () => void;
+}
+
+const FileChooserModal: React.FC<FileProps> = ({ show, onClose }) => {
+  const handleOk = () => {
+    onClose();
+  };
+  const handleCancel = () => {
+    onClose();
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx:14]: We probably wanna add `onCancel` as its own callback in our component as well.

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import Modal from 'antd/lib/modal/Modal';
+
+interface FileProps {
+  show: boolean;
+  onClose: () => void;
+}
+
+const FileChooserModal: React.FC<FileProps> = ({ show, onClose }) => {
+  const handleOk = () => {
+    onClose();
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx:11]: We probably wanna add `onOk` as its own callback in our component as well.

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  };
+  const handleCancel = () => {
+    onClose();
+  };
+  return (
+    <Modal
+      title="Choose a file"
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx:18]: title should probably be prop wit default value in our component as well

File: desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+      open={show}
+      onOk={handleOk}
+      onCancel={handleCancel}
+      okText="Select"
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/reactComponents/FileChooserComponent/FileChooserModal/FileChooserModal.tsx:22]: prop for the okText also

**Review by bjornalm — CHANGES_REQUESTED** [review]
Really nice work! We are on our way, this is great.

File: desktop/libs/indexer/src/indexer/templates/importer.mako [hunk 10]
```diff
@@ -79,6 +79,18 @@ ${ commonheader(_("Importer"), "indexer", user, request, "60px") | n,unicode }
...
+            (function () {    
+              window.createReactComponents('#importerComponents');
+            })();
+          </script>
+          <li>
+            <div>
+              <FileChooserComponentButton data-reactcomponent='FileChooserComponentButton' ></FileChooserComponentButton>
+            </div>
+          </li>
+          !-->
```
Comment (JohanAhlen) [inline desktop/libs/indexer/src/indexer/templates/importer.mako:93]: I guess this one should be removed from the PR. You should keep it for development but let's not push it.

**Review by JohanAhlen — APPROVED** [review]
Nice! :-)

**Review by bjornalm — APPROVED** [review]
Great work!

**Pull request merged**